rt.tf

```diff
@@ -1,54 +1,63 @@
 //Creating route table for private subnets
-resource "aws_route_table" "rt_ngw" {
-  vpc_id = aws_vpc.test-vpc.id
-
+resource "aws_route_table" "private" {
+  vpc_id = aws_vpc.nginx_vpc.id
+  
   tags = {
-    Name = "Premium Auto private RT"
+    Name = "Private route table"
   }
 }
 
-resource "aws_route_table_association" "rt_asso_ngw" {
+resource "aws_route_table_association" "private_association" {
   count = length(var.private_subnet_cidrs)
-  subnet_id = element(aws_subnet.private_subnet[*].id, count.index)
-  route_table_id = aws_route_table.rt_ngw.id
+  subnet_id = aws_subnet.private_subnet[count.index].id
+  route_table_id = aws_route_table.private.id
+ 
 }
 
-//Creating NAT
-resource "aws_nat_gateway" "ngw" {  
-  connectivity_type = "private"
-  count = length(var.private_subnet_cidrs)
-  subnet_id = element(aws_subnet.private_subnet[*].id, count.index)
+resource "aws_eip" "nat_eip" {
+  count = 2
+  vpc = true
+  depends_on = [aws_internet_gateway.igw]
```

Comment:
What can `count` be instead of hardcoded value?

Do we need depends on here, how does EIP depend on IGW?